src/tools-evaluation.md

```diff
@@ -22,7 +38,17 @@ of using Linux to boot into an operating system.
 | [u-root](https://u-root.org)                             | Go       | BSD 3-Clause | ByteDance, Google et al |
 | [uutils/coreutils](http://uutils.github.io/)             | Rust     | MIT          | not for LinuxBoot       |
 
-## kexec implementations
+### kexec implementations
+
+While [kexec itself is a Linux syscall](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/kexec.h),
+it is not a one-shot operation. Loading multiple segments into memory,
+synchronizing and unmounting file systems, and the eventual syscall to
+[reboot](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/reboot.h)
+belong to the procedure. In addition, there are architecture specifics to take
+into account. Thus, there are multiple implementations of kexec, which are Linux
+programs that offer their own interfaces again to pass extra arguments. Besides
+those standalone implementations, there are also specialized boot loaders based
+on kexec that have their own extra logic, such as FreeBSD's kload or petitboot.
```

Comment:
`kexec` primarily refers to the `kexec(8)` shell command implemented in [kexec-tools](https://git.kernel.org/pub/scm/utils/kernel/kexec/kexec-tools.git). It can be used to load a new kernel image into memory using the `kexec_load(2)` system calls. Then it can call the `reboot(2)` system call with `LINUX_REBOOT_CMD_KEXEC`.

There are alternate implementations of `kexec` including:

- FreeBSD kboot
- OpenPower kexec-lite
- u-root kexec